torchbenchmark/models/mobilenet_v2_qat/__init__.py

```diff
@@ -0,0 +1,61 @@
+# Generated by gen_torchvision_benchmark.py
+import torch
+import torch.optim as optim
+import torchvision.models as models
+from torch.quantization import quantize_fx
+from ...util.model import BenchmarkModel
+
+
+class Model(BenchmarkModel):
+    def __init__(self, device=None, jit=False):
+        super().__init__()
+        self.device = device
+        self.jit = jit
+        self.model = models.mobilenet_v2().to(self.device)
+        self.example_inputs = (torch.randn((32, 3, 224, 224)).to(self.device),)
+        self.prep_qat_train()  # config+prepare steps are required for both train and eval
+
+    def prep_qat_train(self):
+        qconfig_dict = {"": torch.quantization.get_default_qat_qconfig('fbgemm')}
+        self.model.train()
+        self.model = quantize_fx.prepare_qat_fx(self.model, qconfig_dict)
+
+    def train(self, niter=3):
+        if self.jit is True:  # torchscript operations should only be applied after quantization operations
+            raise NotImplementedError()
+        optimizer = optim.Adam(self.model.parameters())
+        loss = torch.nn.CrossEntropyLoss()
+        for _ in range(niter):
+            optimizer.zero_grad()
+            pred = self.model(*self.example_inputs)
+            y = torch.empty(pred.shape[0], dtype=torch.long, device=self.device).random_(pred.shape[1])
+            loss(pred, y).backward()
+            optimizer.step()
+
+    def set_eval(self):
```

Comment:
do you need to also override set_train?  i think as-written it could work in CI today but if the order of the tests were changed (eval before train) would it still work?  maybe it would be OK since you construct the model in train mode each time, i guess i forgot if the test harness constructs a new model always.

Reply:
So the short answer is that I don't need to override set_train (that setup is located in the initialization) and that no matter what, there's not really a good way to handle eval before train for quantized models (that I know of) without reinitializing things.

As implemented it works with
i) eval only
ii) train only
ii) train then eval

there's a piece in test_bench that implies that if the order is an issue we  may be able to change the scope of the model to get around that

```Any pytest function that consumes a 'modeldef' arg will invoke this
    automatically, and reuse it for each test that takes that combination
    of arguments within the module.

    If reusing the module between tests isn't safe, change 'scope' parameter.```

but I wasn't sure how to do that for the model or if it would require changing it for the entire repo.

Reply:
IIUC it looks like you are able to run the test_eval and test_train in either order, due to initializing train first in the setup.  I guess that could be thought of as 'train then eval then train' in the worst case, where train happens in 'init', then test_eval runs, then test_train runs.  IF that's true, i don't have any issue with the approach you took.

Reply:
> IIUC it looks like you are able to run the test_eval and test_train in either order, due to initializing train first in the setup. I guess that could be thought of as 'train then eval then train' in the worst case, where train happens in 'init', then test_eval runs, then test_train runs. IF that's true, i don't have any issue with the approach you took.

it will definitely throw an error, the prep_eval step turns it into a quantized form, which doesn't track gradients and doesn't support backprop for the training step. If for example you added self.prep_qat_train() and self.train()to the bottom of the eval function, it won't complete without error.

I'm not sure how to get around that though without entirely reinitializing each time the model is called.

Reply:
i think we can change 
@pytest.fixture(scope='class')
to scope=test, in order to have the fixture construct a new model object for train and then for eval, so it would work in either order.  

that said, i am not sure if i confirmed that understanding before;  also not sure if pytest promises to run tests in a certain order (Relative to how they are listed in the test class, which is train then eval).  If it's at least consistent, i think it's OK to merge.

Reply:
scope='test' seems to be unrecognized, scope='function' works, would that be sufficient?

Reply:
sorry, yea that sounds right.

Reply:
alright, I put together another PR, this most recent one grabbed a bunch of mypy garbage and it was easier for me to make a new PR to fix it: https://github.com/pytorch/benchmark/pull/323

I re ran the test and added the name change recommend by @vkuzo, let me know if there's anything else